backend/cmm_helpers.mli

```diff
@@ -325,24 +325,25 @@ val make_float_alloc :
   expression list ->
   expression
 
-module Flat_suffix_element : sig
-  type t =
-    | Tagged_immediate
-    | Naked_float
-    | Naked_float32
-    | Naked_int32
-    | Naked_int64_or_nativeint
-end
+(** Allocate a closure block. *)
+val make_closure_alloc :
+  mode:Lambda.alloc_mode ->
+  Debuginfo.t ->
+  tag:int ->
+  expression list ->
+  memory_chunk list ->
+  expression
 
-(** Allocate an mixed block of the corresponding tag and shape. Initial values
-    of the flat suffix should be provided unboxed. *)
+(** Allocate an mixed block of the corresponding tag and scannable prefix size.
+    The [memory_chunk] list should give the memory_chunk corresponding to
+    each element from the [expression] list. *)
 val make_mixed_alloc :
   mode:Lambda.alloc_mode ->
   Debuginfo.t ->
   tag:int ->
   value_prefix_size:int ->
-  flat_suffix:Flat_suffix_element.t array ->
   expression list ->
+  memory_chunk list ->
```

Comment:
We've lost some static checking here for the kinds of the flat suffix, which seems unfortunate.  Can either the `Flat_suffix_element` type be reinstated, or checks added to ensure the memory chunk kinds are appropriate for the suffix?  The first of these seems like a better interface to me.

Reply:
Are you sure you don't mean for the prefix ? It would make sense to check that everything in the value prefix is either `Word_val` or `Word_int`, but for the suffix we have very few restrictions, and those restrictions are checked in `alloc_generic_set_fn`.

Reply:
I'll add some checks on the memory_chunks for the mixed block case.

Reply:
I meant for the suffix, in particular to make sure that values are not provided.  We should indeed do the converse for the prefix.

Reply:
I added the check